apps/cli-daemon/src/app/workspace-manager/workspace-manager.service.ts

```diff
@@ -0,0 +1,61 @@
+import fs from 'fs/promises';
+import { parse } from 'path';
+
+import { Directory } from '@angular-cli-gui/shared/data';
+import { NodeJsSyncHost } from '@angular-devkit/core/node';
+import { createWorkspaceHost } from '@angular-devkit/core/src/workspace';
+import { readWorkspace as devKitReadWorkspace } from '@angular-devkit/core/src/workspace/core';
+import { BadRequestException, Injectable } from '@nestjs/common';
+
+import { NOT_VALID_PATH } from './entities';
+
+@Injectable()
+export class WorkspaceManagerService {
+  public async getDirectory(path: string): Promise<Directory[]> {
+    try {
+      const filesAndFolders = await fs.readdir(path, { withFileTypes: true });
```

Comment:
choose a naming convention "folder" or "directory" (most of your code is directory so I think this one)